File: functions/ffmpeg/variants/go/0_flac_get_max_frame_size_0.go
package main

func flacGetMaxFrameSize(blocksize, ch, bps int32) int32 {
    count := int32(16)
    count += ch * ((7 + bps + 7) / 8)
    if ch == 2 {
        count += ((2*bps + 1) * blocksize + 7) / 8
    } else {
        count += (ch * bps * blocksize + 7) / 8
    }
    count += 2
    return count
}


func main(){
    flacGetMaxFrameSize(0,0,0)
}

File: functions/ffmpeg/variants/c/0_flac_get_max_frame_size_0.c
#include <stdint.h>

static int flac_get_max_frame_size(int blocksize, int ch, int bps) 
{
    int count = 16;
    int overhead = (7 + bps + 7) / 8;
    
    count += ch * overhead;
    
    if (ch == 2) {
        count += ((2 * bps + 1) * blocksize + 7) / 8;
    } else {
        count += (ch * bps * blocksize + 7) / 8;
    }
    
    count += 2;
    
    return count;
}


int main(){
    flac_get_max_frame_size(0,0,0);
}
